lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2557,7 +2560,7 @@ HSADispatch::waitComplete() {
         printf("Signal wait returned unexpected value\n");
         exit(0);
     }
-
+    dispose();
```

Comment:
sorry I should be more clear. I'm not referring to remove this line, but the block "if (kernargMemory != nullptr)" below it. (line 2565 - 2568). the logic would be redundant if we call dispose() here.